Automattic/wp-calypso
# State: Add plan transfer action and data layer

This PR introduces the action creator and data layer for transferring a site plan to another user. It is to be used in conjunction with D14186-code. The PR suggests:

* A new action type for issuing a plan transfer request.
* A `transferPlanOwnership()` action creator (+ tests).
* Data layer for handling the plan transfer request (+ tests). 
* Updating the `client/state/sites/plans/test/actions.js` tests to use Jest.

Note: this PR shouldn't be merged before its counterpart D14186-code is merged.

To test:
* Checkout this branch
* ~Apply D14186-code on your sandbox and make sure you're sandboxing the API (instructions for that can be found in the diff itself).~ - this has already landed.
* Hit the following in your dev console: `dispatch( { type: 'SITE_PLAN_OWNERSHIP_TRANSFER', siteId: 1234, newOwnerUserId: 5678 } )`, where `siteId` and `newOwnerUserId` are valid site and user IDs.
* Try with various cases that should fail (invalid site, lack of permissions, non-connected site, non-Jetpack site, etc - a thorough list of these cases can be found at D14186-code).
* Verify tests pass:
  * `npm run test-client client/state/sites/plans/test/actions.js`
  * `npm run test-client client/state/data-layer/wpcom/sites/plan-transfer`
